src/main.js

```diff
@@ -1,19 +1,137 @@
+var viewSavedButton = document.querySelector('.view-saved-button');
+
+
+var formView = document.querySelector('.form-view');
+var homeView = document.querySelector('.home-view');
+var viewSavedView = document.querySelector('.saved-view');
+
+
+var inputCover = document.querySelector('.user-cover');
+var inputTitle = document.querySelector('.user-title');
+var inputFirstDescriptor = document.querySelector('.user-desc1');
+var inputSecondDescriptor = document.querySelector('.user-desc2');
+
+
+var savedCoversSection = document.querySelector('.saved-covers-section');
 
 
-// We've provided a few variables below
-var savedCovers = [
-  new Cover("http://3.bp.blogspot.com/-iE4p9grvfpQ/VSfZT0vH2UI/AAAAAAAANq8/wwQZssi-V5g/s1600/Do%2BNot%2BForsake%2BMe%2B-%2BImage.jpg", "Sunsets and Sorrows", "sunsets", "sorrows")
-];
 var currentCover;
+var savedCovers = [];
+
+
+
+window.addEventListener('load', makeRandomCover);
+
+randomCoverButton.addEventListener('click', makeRandomCover);
+
+createNewBookButton.addEventListener('click', saveCustomBook);
+
+makeCoverButton.addEventListener('click', changeFormView);
+
+homeButton.addEventListener('click', changeHomeView);
+
+viewSavedButton.addEventListener('click', changeViewSavedView);
+
+saveCoverButton.addEventListener('click', saveCurrentCover);
+
+savedCoversSection.addEventListener('dblclick', deleteSavedCover);
 
-// Add your event listeners here 👇
 
 
-// Create your event handlers and other functions here 👇
+function changeFormView() {
+  homeView.classList.add('hidden');
+  formView.classList.remove('hidden');
+  viewSavedView.classList.add('hidden');
+  saveCoverButton.classList.add('hidden');
+  randomCoverButton.classList.add('hidden');
+  homeButton.classList.remove('hidden');
+}
+
+function changeViewSavedView() {
+  randomCoverButton.classList.add('hidden');
+  saveCoverButton.classList.add('hidden');
+  homeButton.classList.remove('hidden');
+  viewSavedView.classList.remove('hidden');
+  homeView.classList.add('hidden');
+  formView.classList.add('hidden');
+}
+
+function changeHomeView() {
+  homeButton.classList.add('hidden');
+  randomCoverButton.classList.remove('hidden');
+  saveCoverButton.classList.remove('hidden');
+  homeView.classList.remove('hidden');
+  formView.classList.add('hidden');
+  viewSavedView.classList.add('hidden');
+}
+
+function makeRandomCover() {
+  currentCover = new Cover(covers[getRandomIndex(covers)], titles[getRandomIndex(titles)], descriptors[getRandomIndex(descriptors)], descriptors[getRandomIndex(descriptors)]);
+  coverImage.src = currentCover.cover;
+  title.innerText = currentCover.title;
+  descriptor1.innerText = currentCover.tagline1;
+  descriptor2.innerText = currentCover.tagline2;
+}
+
+function createNewBook() {
+  covers.push(inputCover.value);
+  titles.push(inputTitle.value);
+  descriptors.push(inputFirstDescriptor.value);
+  descriptors.push(inputSecondDescriptor.value);
+  console.log(inputSecondDescriptor.value);
```

Comment:
Make sure not to leave any console logs in your final version. If you were already planning on removing this, ignore this message.